CMakeLists.txt

```diff
@@ -135,6 +135,10 @@ endif()
 if(MSVC)
   # Required to use M_PI. Must be set before any `#include <cmath>`.
   add_compile_definitions("_USE_MATH_DEFINES")
+
+  # Tablegen outputs from LLVM/MLIR can easily exceed the default 64k section
+  # count limit, so increase to 2^32.
+  add_compile_options("/bigobj")
```

Comment:
This is also set in LLVM: https://github.com/llvm/llvm-project/blob/92ad0397f494a9895385057586cc59a908107f81/llvm/cmake/modules/HandleLLVMOptions.cmake#L666-L669 and in IREE: https://github.com/iree-org/iree/blob/c6056d197230161ea1403e88b5b8784d34e071a2/build_tools/cmake/iree_copts.cmake#L131-L136

Reply:
Which build type gets used here? EXTERNAL or EMBEDDED? The embedded actually uses `HandleLLVMOptions` so would get this for free?

Reply:
External: https://github.com/iree-org/iree/blob/c6056d197230161ea1403e88b5b8784d34e071a2/CMakeLists.txt#L820-L825

Reply:
Got it, maybe we play whack-a-mole this time, and if something like this comes up again we try and see if there's a better way of including `include(HandleLLVMOptions)`.

The TF/mlir-hlo build uses external as well and has already called `include(HandleLLVMOptions)` outside of StableHLO, which likely sets this flag, and calling it twice causes issues. I wonder if theres a good way to check if this has been called previously..

Reply:
(resolve thread if sgty and I'll merge)

Reply:
I don't really have a strong opinion right now. The layering of these LLVM projects is definitely tricky.

In IREE we don't call `include(HandleLLVMOptions)`. We use https://github.com/iree-org/iree/blob/main/build_tools/cmake/iree_llvm.cmake and https://github.com/iree-org/iree/blob/main/build_tools/cmake/iree_copts.cmake.

What we might end up doing with StableHLO is maintaining a parallel CMake build, as we already do for torch-mlir: https://github.com/iree-org/iree/blob/main/compiler/plugins/input/Torch/torch-mlir/CMakeLists.txt. We don't support Bazel for torch-mlir though, so the maintenance burden is a bit lower. With our own build, we could also address https://github.com/openxla/stablehlo/issues/2255 ourselves.